### PR TITLE
Fix cursor bold and italic rendering

### PR DESCRIPTION
**Pull request by PfisterFactor**
## What kind of change does this PR introduce?
- Fix
Addresses #1021

## Did this PR introduce a breaking change? 
_A breaking change includes anything that breaks backwards compatibility either at compile or run time._
Unsure, I changed the Cursor struct's "character" field to store a GridCell instead of a String in order to carry style information.

### CODE REVIEW

File: src/editor/window.rs [hunk 1]
```diff
@@ -63,18 +63,22 @@ impl Window {
...
+            _ => (' '.to_string(), None),
         };
 
         let double_width = match self.grid.get_cell(window_left + 1, window_top) {
             Some((character, _)) => character.is_empty(),
             _ => false,
         };
 
-        (character, double_width)
+        (character.0, character.1, double_width)
```
Comment (Kethku) [inline src/editor/window.rs:81]: nit: I think the local variable character here should be renamed. Maybe grid_cell?

Comment (PfisterFactor) [reply src/editor/window.rs]: I agree, I pushed a commit fixing it 👍

Comment (PfisterFactor) [reply src/editor/window.rs]: Gotcha, I renamed:
Cursor struct: "character" -> "grid_cell"
get_cursor_character -> get_cursor_grid_cell

**Comment by Kethku** [comment]
This is a reasonable fix. Thanks for the contribution :)

**Pull request merged**